inference-engine/src/mkldnn_plugin/nodes/ctc_loss.cpp

```diff
@@ -0,0 +1,348 @@
+                THROW_IE_EXCEPTION << _logPrefix << " has nullable input data";
+            auto prc = data->getTensorDesc().getPrecision();
+            if (prc == Precision::BF16)
+                prc = Precision::FP32;
+            config.inConfs[i].desc = TensorDesc(prc,
+                data->getTensorDesc().getDims(),
+                TensorDesc::getLayoutByDims(data->getTensorDesc().getDims()));
+        }
+
+        auto logitsData = layer->insData[0].lock();
+        if (logitsData == nullptr)
+            THROW_IE_EXCEPTION << _logPrefix << " has nullable logits data.";
+        DataConfig outConfig;
+        auto& outDims = layer->outData[0]->getTensorDesc().getDims();
+        outConfig.desc = TensorDesc(logitsData->getTensorDesc().getPrecision(),
+            outDims,
+            TensorDesc::getLayoutByDims(outDims));
+        config.outConfs.push_back(outConfig);
+        config.dynBatchSupport = false;
+
+        confs.push_back(config);
+    }
+
+    StatusCode execute(std::vector<Blob::Ptr>& inputs,
+                       std::vector<Blob::Ptr>& outputs,
+                       ResponseDesc *resp) noexcept override {
+        switch (inputs[1]->getTensorDesc().getPrecision()) {
+            case Precision::I32: {
+                return processData<PrecisionTrait<Precision::I32>::value_type>(inputs, outputs, resp);
+            }
+            case Precision::I64: {
+                return processData<PrecisionTrait<Precision::I64>::value_type>(inputs, outputs, resp);
+            }
+            default: {
+                if (resp) {
+                    std::string errorMsg = _logPrefix + " does not support precision '"
+                            + std::string(inputs[1]->getTensorDesc().getPrecision().name()) + "'";
+                    errorMsg.copy(resp->msg, sizeof(resp->msg) - 1);
+                }
+                return GENERAL_ERROR;
+            }
+        }
+    }
+
+protected:
+    template<typename I1>
+    StatusCode processData(
+                std::vector<Blob::Ptr>& inputs,
+                std::vector<Blob::Ptr>& outputs,
+                ResponseDesc* resp) noexcept {
+        switch (inputs[2]->getTensorDesc().getPrecision()) {
+            case Precision::I32: {
+                return processData<I1, PrecisionTrait<Precision::I32>::value_type>(inputs, outputs, resp);
+            }
+            case Precision::I64: {
+                return processData<I1, PrecisionTrait<Precision::I64>::value_type>(inputs, outputs, resp);
+            }
+            default: {
+                if (resp) {
+                    std::string errorMsg = _logPrefix + " does not support labels precision '"
+                            + std::string(inputs[2]->getTensorDesc().getPrecision().name()) + "'";
+                    errorMsg.copy(resp->msg, sizeof(resp->msg) - 1);
+                }
+                return GENERAL_ERROR;
+            }
+        }
+    }
+
+    template<typename I1, typename I2>
```

Comment:
I am totally against templating such huge functions like `processingData`. We just increasing our binary size for nothing.

My recommendation is to support only I32 for `logitsLength` , `labels`, `labelsLength` and blankIndex (in other words lets create only I32 config for this inputs). The plug-in will handle data conversion automatically. 
According to the spec it is highly unlikely what this inputs can contain elements that are greater when INT_MAX.

Reply:
Fixed